static-instrumenter/agent-instrumenter/src/integrationTest/java/JarTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+final class JarTest {
+
+  private static final String INSTRUMENTATION_MAIN =
+      "io.opentelemetry.contrib.staticinstrumenter.agent.main.Main";
+  private static final String APP_MAIN = "org.example.SingleGetter";
+
+  @TempDir public Path outPath;
+
+  @Test
+  void testSampleJar() throws IOException, InterruptedException {
+    Path agentPath = getPath("opentelemetry-static-agent.jar");
+    Path noInstAgentPath = getPath("opentelemetry-no-inst-agent.jar");
+    Path appPath = getPath("app.jar");
```

Comment:
Is [this](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/static-instrumenter/test-app) where the app code comes from? Can you add a comment explaining that?